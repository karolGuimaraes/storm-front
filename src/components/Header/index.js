import React, {useState} from 'react';
import styles from './styles.module.css';
import logo from '../../assets/symbol.svg';
import SecurityIcon from '@material-ui/icons/Security';
import PersonIcon from '@material-ui/icons/Person';
import SearchIcon from '@material-ui/icons/Search';
import TuneIcon from '@material-ui/icons/Tune';
import HomeIcon from '@material-ui/icons/Home';
import SettingsIcon from '@material-ui/icons/Settings';
import PowerSettingsNewIcon from '@material-ui/icons/PowerSettingsNew';
import { Drawer } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import DateRangeIcon from '@material-ui/icons/DateRange';
import HdrStrongIcon from '@material-ui/icons/HdrStrong';
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  paper: {
    backgroundColor: '#F5F5F5',
    width: '500px',
  }
}));

export default function Header(props) {
  const classes = useStyles();
  const [openSiderBar, setOpenSiderBar] = useState(false);

  const siderBar = (open) => {
    setOpenSiderBar(open);
  }

  return (
    <header className={styles.header}>
      <Drawer open={openSiderBar} anchor="right" classes={{ paper: classes.paper }}>
        <div className={styles.headerSiderBar}>
          <div className={styles.sideBarLeft}>
            <TuneIcon style={{ color: '#c1c1c1' }} />
            <span>FILTROS</span>
          </div>
          <button onClick={() => siderBar(false)} className={styles.btnClose}>
            <CloseIcon />
          </button>
        </div>
        <div>
          <div className={styles.content}>
            <p>
              Utilize os filtros abaixo para refinar os resultados da tabela,
              clique no botão APLICAR para salvar as alterações.
            </p>
          </div>
          <div className={styles.filter}>
            <DateRangeIcon className={styles.iconSideBar} />
            <span>TODAS AS DATAS DE INCLUSÃO</span>
            <ArrowDropDownIcon className={styles.iconSideBar} />
          </div>
          <div className={styles.filter}>
            <DateRangeIcon className={styles.iconSideBar} />
            <span>TODAS AS DATAS DE ALTERAÇÃO</span>
            <ArrowDropDownIcon className={styles.iconSideBar} />
          </div>
          <div className={styles.filter}>
            <HdrStrongIcon className={styles.iconSideBar} />
            <span>ATIVOS E INATIVOS</span>
            <ArrowDropDownIcon className={styles.iconSideBar} />
          </div>

          <div className={styles.btnApply}>
            <button>
              APLICAR
            </button>
          </div>
        </div>
      </Drawer>

      <div className={styles.headerLeft}>
        <button className={styles.logo}>
          <img src={logo} alt='logo'/>
        </button>
        <span className={styles.separator}></span>
        <button className={styles.btnDefender}>
          <SecurityIcon />
        </button>
        <button className={styles.btnPerson2}>
          <PersonIcon />
        </button>

        <div className={styles.containerSearch}>
          <input className={styles.input} placeholder="Pesquisar..."/>
          <SearchIcon className={styles.icon}/>
        </div>
      </div>

      <div className={styles.headerRight}>
        <button className={styles.btnTune} onClick={(event) => siderBar(true)}>
          <TuneIcon />
        </button>
        <button className={styles.btnPerson}>
          <PersonIcon className={styles.btnIcon}/>
          <span className={styles.addUserText} >INCLUIR USUÁRIO</span>
        </button>
        <span className={styles.separator}></span>
        <button className={styles.icon}>
          <HomeIcon />
        </button>
        <button className={styles.icon}>
          <SettingsIcon />
        </button>
        <button className={styles.icon}>
          <PowerSettingsNewIcon />
        </button>
      </div>
    </header>
  )
}