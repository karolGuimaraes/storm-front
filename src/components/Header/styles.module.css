.header {
  display: flex;
  justify-content: space-between;
  align-items: center;

  /* box-shadow: 0 4px 4px -2px #c1c1c1;
  -moz-box-shadow: 0 4px 4px -2px #c1c1c1;
  -webkit-box-shadow: 0 4px 4px -2px #c1c1c1; */
  background-color: #F5F5F5;

  height: 100px;
  width: 100%;
}

.logo {
  border: none;
  background-color: #F5F5F5;
}

.pipe {
  height: 28px;
  background-color: #c1c1c1;
  width: 2px;
  margin: 0 25px;
}

.headerLeft {
  display: flex;
  justify-content: center;
  align-items: center;
}

.headerRight {
  display: flex;
    justify-content: center;
    align-items: center;

    margin-right: 50px;
}

.btnLogo {
  padding: 15px;
  margin-left: 50px;
  border-radius: 16px;
  border: none;
}

.btnTune {
  padding: 15px;
  background-color: #fff;
  border: none;
  border-radius: 4px;
}

.btnInclude {
  padding: 15px;
  margin-left: 25px;
  background-color: #a73b5f;
  border: none;
}

.btnDefender {
  padding: 15px;
  background-color: #fff;
  border: none;
  border-radius: 4px;
}

.btnPerson {
  padding: 15px;
  background: #D83367;
  border-radius: 4px;
  color: #fff;
  border: none;
}

.input {
 border: none; 
 background-color: #F5F5F5;
}

.btnPerson2 {
  padding: 15px;
  background-image: linear-gradient(#FF8700, #D83367);
  border-radius: 4px;
  color: #fff;
  border: none
}

.containerSearch {
  margin-left: 40px;
  box-shadow: 0 2px 3px -2px #c1c1c1;
  -moz-box-shadow: 0 4px 4px -2px #c1c1c1;
  -webkit-box-shadow: 0 4px 4px -2px #c1c1c1;
}

.addUserText {
  margin-left: 10px;
  font-size: 14px;
  color: white;
}

.icon {
  color: #666666;
  background-color: #F5F5F5;
  font-weight: 400;
  border: none
}