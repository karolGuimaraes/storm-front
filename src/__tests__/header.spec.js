import React from 'react';
import Header from '../components/Header';
import { create } from "react-test-renderer";
import { shallow, render } from 'enzyme';

describe("Header Component Test", () => {
  it("renders correctly", () => {
    const header = create(<Header />);
    expect(header.toJSON()).toMatchSnapshot();
  });
});