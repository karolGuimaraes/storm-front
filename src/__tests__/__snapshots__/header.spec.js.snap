// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Test renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="headerLeft"
  >
    <button
      className="logo"
    >
      <img
        alt="logo"
        src="symbol.svg"
      />
    </button>
    <span
      className="separator"
    />
    <button
      className="btnDefender"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11v8.8z"
        />
      </svg>
    </button>
    <button
      className="btnPerson2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </button>
    <div
      className="containerSearch"
    >
      <input
        className="input"
        placeholder="Pesquisar..."
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
  </div>
  <div
    className="headerRight"
  >
    <button
      className="btnTune"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17v2h6v-2H3zM3 5v2h10V5H3zm10 16v-2h8v-2h-8v-2h-2v6h2zM7 9v2H3v2h4v2h2V9H7zm14 4v-2H11v2h10zm-6-4h2V7h4V5h-4V3h-2v6z"
        />
      </svg>
    </button>
    <button
      className="btnPerson"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root btnIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
      <span
        className="addUserText"
      >
        INCLUIR USUÁRIO
      </span>
    </button>
    <span
      className="separator"
    />
    <button
      className="icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </button>
    <button
      className="icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
          transform="scale(1.2, 1.2)"
        />
      </svg>
    </button>
    <button
      className="icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13 3h-2v10h2V3zm4.83 2.17l-1.42 1.42C17.99 7.86 19 9.81 19 12c0 3.87-3.13 7-7 7s-7-3.13-7-7c0-2.19 1.01-4.14 2.58-5.42L6.17 5.17C4.23 6.82 3 9.26 3 12c0 4.97 4.03 9 9 9s9-4.03 9-9c0-2.74-1.23-5.18-3.17-6.83z"
        />
      </svg>
    </button>
  </div>
</header>
`;

exports[`Test Header Component should  1`] = `
<header
  className="header"
>
  <div
    className="headerLeft"
  >
    <button
      className="logo"
    >
      <img
        alt="logo"
        src="symbol.svg"
      />
    </button>
    <span
      className="separator"
    />
    <button
      className="btnDefender"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11v8.8z"
        />
      </svg>
    </button>
    <button
      className="btnPerson2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </button>
    <div
      className="containerSearch"
    >
      <input
        className="input"
        placeholder="Pesquisar..."
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
  </div>
  <div
    className="headerRight"
  >
    <button
      className="btnTune"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17v2h6v-2H3zM3 5v2h10V5H3zm10 16v-2h8v-2h-8v-2h-2v6h2zM7 9v2H3v2h4v2h2V9H7zm14 4v-2H11v2h10zm-6-4h2V7h4V5h-4V3h-2v6z"
        />
      </svg>
    </button>
    <button
      className="btnPerson"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root btnIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
      <span
        className="addUserText"
      >
        INCLUIR USUÁRIO
      </span>
    </button>
    <span
      className="separator"
    />
    <button
      className="icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </button>
    <button
      className="icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
          transform="scale(1.2, 1.2)"
        />
      </svg>
    </button>
    <button
      className="icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13 3h-2v10h2V3zm4.83 2.17l-1.42 1.42C17.99 7.86 19 9.81 19 12c0 3.87-3.13 7-7 7s-7-3.13-7-7c0-2.19 1.01-4.14 2.58-5.42L6.17 5.17C4.23 6.82 3 9.26 3 12c0 4.97 4.03 9 9 9s9-4.03 9-9c0-2.74-1.23-5.18-3.17-6.83z"
        />
      </svg>
    </button>
  </div>
</header>
`;
